File: php/src/net_tcp6_receiver.php
<?php

set_time_limit(0);

const RECEIVER_ADDRESS = "::1";
const RECEIVER_PORT = 10000;
const RECEIVED_MESSAGES_MAX = 10;
const BUFFER_SIZE = 4096;

echo "TCP IPv6 stream socket receiver".PHP_EOL;

$socket = @socket_create(AF_INET6, SOCK_STREAM, SOL_TCP);
if ($socket === false)
{
  exit("- socket_create() failed: ".socket_strerror(socket_last_error()).PHP_EOL);
}
echo "- socket created".PHP_EOL;

$bound = @socket_bind($socket, RECEIVER_ADDRESS, RECEIVER_PORT);
if ($bound === false)
{
  exit("- socket_bind() failed: ".socket_strerror(socket_last_error($socket)).PHP_EOL); 
}
echo "- socket bound on ".RECEIVER_ADDRESS.":".RECEIVER_PORT.PHP_EOL;

$listening = @socket_listen($socket, RECEIVED_MESSAGES_MAX);
if ($listening === false)
{
  exit("- socket_listen() failed: ".socket_strerror(socket_last_error($socket)).PHP_EOL); 
}
echo "- socket is listening for max ".RECEIVED_MESSAGES_MAX." messages".PHP_EOL;

$stop = false;
while ($stop == false)
{
  $msgsock = @socket_accept($socket);
  if ($msgsock === false)
  {
    exit("- socket_accept() failed: ".socket_strerror(socket_last_error($socket)).PHP_EOL);
  }
  echo "- socket accepted request".PHP_EOL;

  $peer_address = "";
  $peer_port = 0;

  $peer = @socket_getpeername($msgsock, $peer_address, $peer_port);
  if ($peer === false)
  {
    exit("- socket_getpeername() failed: ".socket_strerror(socket_last_error($msgsock)).PHP_EOL);
  }
  echo "- peer connect from ".$peer_address.":".$peer_port.PHP_EOL;

  $msg = "";

  $received_length = @socket_recv($msgsock, $msg, BUFFER_SIZE, 0);
  if ($received_length === false)
  {
    exit("- socket_recv() failed: ".socket_strerror(socket_last_error($msgsock)).PHP_EOL);
  }
  echo "- message ".$received_length."B received".PHP_EOL;
  echo "|".$msg."|".PHP_EOL;
      
  if ($msg == "stop")
  {
    // received message "stop" to stop the receiver
    $stop = true;
  }
   
  socket_close($msgsock); 
  echo "- socket closed request".PHP_EOL;
}

socket_close($socket);
echo "- socket closed".PHP_EOL;

?>

File: python/src/net_tcp6_receiver.py
import socket

RECEIVER_ADDRESS = "::1"
RECEIVER_PORT = 10000
RECEIVED_MESSAGES_MAX = 10
BUFFER_SIZE = 4096

print("TCP IPv6 stream socket receiver")

receivingSocket = socket.socket(family=socket.AF_INET6, type=socket.SOCK_STREAM)
print("- socket created")

receiver = (RECEIVER_ADDRESS, RECEIVER_PORT);

receivingSocket.bind(receiver)
print("- socket bound on [" + RECEIVER_ADDRESS + "]:" + str(RECEIVER_PORT))

receivingSocket.listen(RECEIVED_MESSAGES_MAX)
print("- socket is listening for max " + str(RECEIVED_MESSAGES_MAX) + " messages")

stop = False
while (stop == False):
    (msgsock, peer) = receivingSocket.accept()
    print("- socket accepted request")

    (peer_address, peer_port, peer_object3, peer_object4) = peer
    (local_address, local_port, local_object3, local_object4) = msgsock.getsockname()
    print("- peer connect from [" + peer_address + "]:" + str(peer_port) + " on [" + local_address + "]:" + str(local_port))

    buffer = msgsock.recv(BUFFER_SIZE)
    received_length = len(buffer)
    msg = buffer.decode()
    print("- message " + str(received_length) + "B received")
    print("|" + msg + "|")

    if msg == "stop":
        stop = True

    msgsock.close()
    print("- socket closed request")

receivingSocket.close()
print("- socket closed")
